### DISCUSSION TITLE
How to connect two objects parallel side-by-side in  a top-down diagram? D2 seems to put one above the other...

### DISCUSSION THREAD

**Discussion by HariSekhon** [terrastruct/d2#1300] [General]
[My Playground code](https://play.d2lang.com/?script=rFZdb9s2FH3nrzirEbTFJktR1wwlkAFp2ixB0jaotz4FKCjqWiJMkQJJWfOK_PeBkuzZQzJsQ_SQiLxf51zyXHmG79LOu7RQJiWzRpkjSUJNDSHPMjZjM-Csi13muBROYUGr2pq4_U4E4siz_FWSvU6yE2QZz3_ir0_wfXacZXixEOEHZCf4IDaD28sx3Vo1PPjTnPvxb3-acwp8qTSFTUunZT761SG0nqdppULdFXNpmzQiGAGk75SonGh8InxybksaY26UJOOJwxNBSGmbVpiNMtU-eNxcnb__uHiPWHKMu1piY7vnjtD56N1sIG1J282etLQNIVhIawzJgF6FGg0hplNmReWVgTAlbBvFIrTewJMpo8uSqCyEXMXwmnQLH4gcQq08rIMNNbmx3BXartDK14cd6Pt-rociygx9UGavFWzG2AynT_hgqP735xodChAcDAgBBA_gEwwStHAgNFDwoKdHw3ZTiKO0vWEsqKCJ4xsDBo1z3FhR4q3QwkhyHp9McuuoUZ4YEDXOEWybSDKBHAMGPXNE6cRVFN18N3c4fsx2uzvxcATXEbtnbJqUY31DobduNS6AOIL47ujiys8DOSd8cJ0Mw_mR92SCEtof5RfZmzyZcsz9uhqSTOhUI6qI_34AY52oaFtmYv3s8gwfx2AqsRh97szSOlx3BTlDgfyduSXnlQ9kAr7EcUr-2b8GK_7oHB3lF1PyozxbkFsrGd_O49w-yi9-IUNO6Ggbvf6Jibai_FpMJ_UAn4OTvDMvLs9unf19gxS_OkFLtboz19ZUSPGBgtA3b1-OZGaISiIob54HCFTWlqgiMiWHotgWHXjD20ntgxJjy4ztkcD3Ksg6qtUagl3GtGidLTsZ4FuSaqnkkMJDDYMDqxjZ10rWMeaB1pbTuJo3ysSXhtyo5KZK3fRd8anofTpdhZS08EHJJAJPRuDKVPPWVBPZ_1PAmtZRs6tRizZ29mmTruxTwwzjsT-aVAtTeilamksjl3NlU20r69MmXo9EF49fxnvGdHHM8W1QND-8mvdMF_mjNrbT_aS_4SJPvpONRazjFJl-W3BksWjnyXmO3-K_Ie4_T41Xr98kQ5aJ3AG1qcLXCQaSn6GL4wf2cjaLFiQJBq5fPn--HVoSzVsSusj3lwdp9tgfUt-SjJH7AfFbtSJqhVZrKo85ronas7jA8YEp3zfl7C-ge9GHy3yiwf4MAAD__w%3D%3D&theme=200&layout=dagre&sketch=0&) has two load balancers side-by-side:

![d2](https://user-images.githubusercontent.com/2211051/236589767-6ab0bafe-825b-43d0-8fb9-bc481983d226.png)

but when I connect the two like `lb1 -- lb2` to represent a parallel HA connection between the two, it puts one of the LBs above the other like this:

![d2 (1)](https://user-images.githubusercontent.com/2211051/236589796-144d5d67-ec2b-44ea-9b83-1cdfeb123010.png)

Any idea how I can keep the load balancers connected side-by-side?

**Comment by westmark** [comment]
I have a similar issue [here](https://play.d2lang.com/?script=1FKxasNADN3vK14JHeMzhi43dGkpzVbqDwiuI2xDfA6SYgjF_14sX-oMgc6dTu-ddHp6ug0e_FnYf3XRUxxxKLDdaks9ochzt3HLiAGlnfh2QMudBLx_7kqUF1HqjQXMooCurxoyYt5GQKt6kuD9jCRTYq5E-VxrVg-9V6rbx-JNiEdiyWRsrNTEBTzlhpZVJzg5QKxvSP1_iX0XlRqu5v96vcRu5ZJ4bJ9TwRrdlt59b07dZwcS7aIR2b2eL0OMVOvAbnLuJjngdQXm16m68HA8BnwswVXRP_IyjfCnnZP7CQAA__8%3D&theme=200&layout=dagre&sketch=0&)

![image](https://user-images.githubusercontent.com/793082/236683185-3d5a9eb4-8d67-42ef-ac4e-f1a3ae635480.png)


Would really like to force the outgoing arrow from `source.system_integration` to emerge on the right-hand side of the box, and enter the other box on the left side.

(Sidenote: D2 is awesome, so happy I found it!)

**Comment by HariSekhon** [comment]
It's not clear to me how to create something even as simply balanced as this:

![image](https://github.com/terrastruct/d2/assets/2211051/7118412a-29f5-4d8f-ac43-b8ffc00ea4dc)

**Comment by alixander** [comment]
sorry for the delay. 

this actually is a limitation of dagre. https://github.com/mermaid-js/mermaid/issues/3723

we've previously tried to shim this but it was not that robust: https://github.com/terrastruct/d2/pull/1120

it should be possible in ELK (and TALA of course). Not sure how married to dagre you are @HariSekhon @westmark 

we'll look into this again next sprint. hopefully come out with a solution this time.

**Reply by westmark** [reply]
@alixander  Not married at all since I just started out. :) I tried ELK and ended up with this 
![image](https://github.com/terrastruct/d2/assets/793082/5921554a-4f27-4002-97b5-1213ff37c762)

It is not clear to me if and how we can modify ELK's [layout configuration](https://www.eclipse.org/elk/reference/groups.html). Is there a way? I have no idea if it would help (and it would of course have a lock in effect) but I'd be happy to experiment

**Reply by HariSekhon** [reply]
I usually try switching to ELK temporarily for each diagrams to see how it renders but I find it often more blocky and less natural so usually end up going back to dagre as it looks better IMO...

I'll look at ELK again...
